pkg/tsdb/jaeger/client.go

```diff
@@ -106,6 +114,83 @@ func (j *JaegerClient) Operations(s string) ([]string, error) {
 	return operations, err
 }
 
+func (j *JaegerClient) Search(query *JaegerQuery) ([]TraceResponse, error) {
+	jaegerURL, err := url.Parse(j.url)
+	if err != nil {
+		return []TraceResponse{}, fmt.Errorf("failed to parse Jaeger URL: %w", err)
+	}
+	jaegerURL.Path = "/api/traces"
+
+	var queryTags string
+	if query.Tags != "" {
+		tagMap := make(map[string]string)
+		decoder := logfmt.NewDecoder(strings.NewReader(query.Tags))
+		for decoder.ScanRecord() {
+			for decoder.ScanKeyval() {
+				key := decoder.Key()
+				value := decoder.Value()
+				tagMap[string(key)] = string(value)
+			}
+		}
+
+		marshaledTags, err := json.Marshal(tagMap)
+		if err != nil {
+			return []TraceResponse{}, fmt.Errorf("failed to convert tags to JSON: %w", err)
+		}
+
+		queryTags = string(marshaledTags)
+	}
+
+	queryParams := map[string]string{
+		"service":     query.Service,
+		"operation":   query.Operation,
+		"tags":        queryTags,
+		"minDuration": query.MinDuration,
+		"maxDuration": query.MaxDuration,
+	}
+
+	urlQuery := jaegerURL.Query()
+	if query.Limit > 0 {
+		urlQuery.Set("limit", fmt.Sprintf("%d", query.Limit))
+	}
+
+	for key, value := range queryParams {
+		if value != "" {
+			urlQuery.Set(key, value)
+		}
+	}
+
+	jaegerURL.RawQuery = urlQuery.Encode()
```

Comment:
Are we forgetting to add time range params? In the picture on the right is query run through frontend and it has more results when i select last 7 days (as I have been running docker block for a while). So I am thinking if time range params are missing here. 

<img width="1508" alt="image" src="https://github.com/user-attachments/assets/daf0a8d8-a5e1-4046-939f-dd8dbf94b9df" />

Reply:
very good catch! updated